### ISSUE TITLE
exported class constructor injection through webpack/babel

### ISSUE THREAD

**Issue by esion** [olov/ng-annotate#213]
Hello,

TL;DR: Is it possible to support this kind of output ?

~~~
var FoobarController =
/*@ngInject*/
exports.FoobarController = function FoobarController(FoobarHelper) {
  _classCallCheck(this, FoobarController);

  this.foo = FoobarHelper.process('bar');
};
~~~

---

Here is my use case, my project is written in es6 and I'm using webpack + babel.
So the initial code is:

~~~
export class FoobarController {
  /*@ngInject*/
  constructor(FoobarHelper) {
    this.foo = FoobarHelper.process('bar');
  }
}
~~~

I know there's some tricks to fix this but I want to be consistent and easy to read/understand. For example this code works perfectly:

~~~
export class FooController {
  /*@ngInject*/
  constructor(FoobarHelper) {
    this.FoobarHelper = FoobarHelper;
  }

  process() {
    return this.FoobarHelper.process('bar');
  }
}
~~~ 

because it generates the following:

~~~
var FooController = exports.FooController = (function () {
  /*@ngInject*/

  function FooController(FoobarHelper) {
    var _this = this;

    _classCallCheck(this, FooController);

    this.FoobarHelper = FoobarHelper;
  }
  FooController.$inject = ["FoobarHelper"];

  _createClass(FooController, [{
    key: 'process',
    value: function process() {
      return this.FoobarHelper.process('bar');
    }
  }]);

  return FooController;
})();
~~~

I'm not sure wich part is involved, may be it is due to babel, I saw this [PR](https://github.com/babel/babel/pull/1184/files)  that adds `var myClass = function myClass(...` or may be webpack sees a simplification when it there is a simple function within a closure. Unfortunately I do not mastering it yet, any help will be very appreciate :)

related to #175

**Comment by olov** [comment]
You should use `"ngInject"` rather than `/*@ngInject*/ (everywhere and for this example in particular), like so:

```js
export class FoobarController {
  constructor(FoobarHelper) {
    "ngInject";
    this.foo = FoobarHelper.process('bar');
  }
}
```

Please re-open if you still have problems.

**Comment by esion** [comment]
thank you for your time & response. This solution is pretty nice.

(sorry for late feedback). 
And thank you for this library.
